hooks/pre/31_install_scl_packages.rb

```diff
@@ -1,8 +1,16 @@
-# Working around https://tickets.puppetlabs.com/browse/PUP-2169
-if el7?
-  packages = []
-  packages << 'rh-postgresql12-postgresql-server' if local_postgresql?
-  packages << 'rh-redis5-redis' if local_redis?
+# Working around https://tickets.puppetlabs.com/browse/PUP-10548
+packages = []
+
+if facts[:selinux]
+  if el7?
+    packages << 'rh-postgresql12-postgresql-server' if local_postgresql?
+    packages << 'rh-redis5-redis' if local_redis?
+  end
+
+  packages << 'foreman-selinux' if foreman_server?
+  packages << 'katello-selinux' if katello_enabled?
+  packages << 'candlepin-selinux' if katello_enabled?
   packages << 'pulpcore-selinux' if pulpcore_enabled?
-  ensure_packages(packages, 'installed')
 end
+
+ensure_packages(packages, 'installed') unless packages.empty?
```

Comment:
`ensure_packages` already has the shortcircuit, no need to repeat this:
https://github.com/theforeman/foreman-installer/blob/79f509685e75141038f03e0505018ba7e0a920de/hooks/boot/01-kafo-hook-extensions.rb#L18-L19

```suggestion
ensure_packages(packages, 'installed')
```